chainsaw-demo/src/bin/chainsaw-demo-http/main.rs

```diff
@@ -0,0 +1,30 @@
+use crate::config::get_configuration;
+use axum::{routing, Router};
+use chainsaw_demo::{logging, Result};
+use tokio::signal;
+
+mod config;
+mod greeter;
+
+#[tokio::main]
+async fn main() -> Result<()> {
+    let configuration = get_configuration()?;
+
+    let subscriber = logging::new_subscriber(configuration.log.level)?;
+    logging::set_global_logger(subscriber)?;
+
+    let http_addr = configuration.http.serve_addr();
+    let http_router = Router::new()
+        .route("/:name/:surname", routing::get(greeter::greeter))
+        .layer(logging::http_trace_layer());
+    let http = axum::Server::bind(&http_addr).serve(http_router.into_make_service());
+
+    tracing::info!(?http, "Revving up HTTP Chainsaw!");
+    tokio::spawn(http);
+
+    signal::ctrl_c()
```

Comment:
I wondered if we wanted to listen for multiple signals here, or rather SIGTERM as well as SIGINT.

Reply:
Yeah, `SIGINT` is used here because Tokio provides that nice utility function for waiting on the signal. They provide some building blocks for signal handling but no easy one-liners for handling signals other than `SIGINT`.